src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static class LambdaUtilities
```

Comment:
Most of the helpers are actually used by both the compiler and the IDE. The file is linked to the IDE.